### PR TITLE
add: new function "penetrate" simulating a remote hacking

### PR DESCRIPTION
**Pull request by EV9H**
Module requirement:  
> pip install pyfiglet 
> pip install rich
(as well as "os" and "keyboard" since used in "hackertype.py")

How it goes: 
step 1 - show header 
step 2 - a loading bar 
step 3 - uses "hackertype.py" by Bruce to simulating hacking codes 
step 4 - use "esc" key to log out

TODO: 
- Decide whether to keep "loadingbar_by_input" function
- Use a substitute code snippet for step 3

### CODE REVIEW

File: src/team13LoremIpsum/penetrate.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        pass
+    con.print(
+        "Admin privileges acquired... Ready to implant virus", style="bold red")
+    hackertype.hackertype('python')
+    print("\n")
+    con.print("\nLogged Out... Successfully implanted", style="bold green")
+    sleep(1)
+
+
+penetrate()
```
Comment (bxw201) [inline src/team13LoremIpsum/penetrate.py:83]: Could we move this into a `if __name__ == '__main__'` block?

File: src/team13LoremIpsum/penetrate.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import random
+import math
+from time import sleep
+
+import pyfiglet
+from rich.console import Console
+
+import hackertype
+
+# Global Var
```
Comment (bxw201) [inline src/team13LoremIpsum/penetrate.py:12]: Is there any way we can avoid global variables here? I think it leaves the potential for unintended side effects. We can just pass these in as parameters.

File: src/team13LoremIpsum/penetrate.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                                  ) + "%" + bar + str("{:.2f}".format(increment)) + "/s", end='\r', flush=True)
+
+
+def penetrate():
+    con = Console()
+    show_header(con)
+    updatebar(con)
+    con.print(
+        "Admin privileges acquired... Ready to implant virus code", style="bold red")
+    hackertype.hackertype('python')
```
Comment (bxw201) [inline src/team13LoremIpsum/penetrate.py:51]: Should we let the user specify which language they want to be emulating?

File: src/team13LoremIpsum/penetrate.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            increment = random.random()
+            if math.floor(progress) < math.floor(progress + increment):
+                bar = "[" + math.floor(progress+increment) * loading_bar_char + \
+                    (10-math.floor(progress+increment)) * "_" + "]"
+            progress += increment
+            print("{:.2f}".format(min((progress*10), 100)
+                                  ) + "%" + bar + str("{:.2f}".format(increment)) + "/s", end='\r', flush=True)
+
+
+def penetrate():
```
Comment (bxw201) [inline src/team13LoremIpsum/penetrate.py:45]: By my last comment, I meant we should pass in the name of the programming language as a parameter. The reason being that it then matches the requirements the prof lays out. To actually prompt for info, we can stick that in the `__main__` file (as a command line parameter).

Comment (EV9H) [reply src/team13LoremIpsum/penetrate.py]: oh right sorry, I fixed it just now

**Comment by bxw201** [comment]
Merged a new version of this pr

**Pull request closed without merging**